android/titanium/src/java/org/appcelerator/titanium/util/TiUrl.java

```diff
@@ -273,21 +273,21 @@ public static String absoluteUrl(String defaultScheme, String url, String baseUr
 			} else {
 				URI uri = new URI(url);
 				if (uri.getScheme() != null) {
-					// the url already has a scheme, so we ignore the
-					// baseUrl completely.
+					// The URL already has a scheme. So, we ignore the base URL completely.
 					combined = url;
-				} else if (baseUrl.endsWith(PATH_SEPARATOR) && url.startsWith(PATH_SEPARATOR)
-						   && !baseUrl.equals("file://")) {
-					if (baseUrl.length() == 1 && url.length() == 1) {
-						combined = PATH_SEPARATOR;
-					} else if (baseUrl.length() == 1) {
-						combined = url;
-					} else if (url.length() == 1) {
-						combined = baseUrl;
+				} else if (url.startsWith(PATH_SEPARATOR)) {
+					// The URL is an absolute path. Not relative to base URL.
+					// Note: A "file:///" URL needs 3 slashes to reference localhost.
+					if (defaultScheme != null) {
+						combined = defaultScheme + PATH_SEPARATOR;
+						if (defaultScheme.equals("file:")) {
```

Comment:
Could we use `TiFileFactory.FILE_URL_SCHEME` here? Or even `FILE_URL_PREFIX`

Reply:
Yeah, we have several "file:" string literals in the "TiURL.java" code.
I also don't like how "TiURL.java" expects a scheme string to end with a trailing colon ':' either. That's not part of the scheme. That's the character which separates it from the scheme from the path.

I didn't feel like mucking with this part and left it alone.